typing/typedecl.ml

```diff
@@ -668,7 +671,7 @@ let check_well_founded_decl env loc path decl to_check =
 
 (* Check for ill-defined abbrevs *)
 
-let check_recursion env loc path decl to_check =
+let check_recursion ~orig env loc path decl to_check =
```

Comment:
Could you make it `orig_env`? We nearly always make environments explicit with their name, and I find the code easier to follow if it is explicit here too.

Reply:
Done.